syft/formats/syftjson/to_syft_model.go

```diff
@@ -78,6 +81,14 @@ func toSyftRelationships(doc *model.Document, catalog *pkg.Catalog, relationship
 	return out
 }
 
+func toSyftSource(s model.Source) *source.Source {
+	newSrc := &source.Source{
+		Metadata: *toSyftSourceData(s),
+	}
+	newSrc.SetID()
```

Comment:
This method will generate a consistent ID across encode/decode

Reply:
We need to call `SetID` since `id` is a private field. Even if I carried doc into this function signature `newSrc.id = doc.Source.ID`

It's important that our `SetID` call is consistent across serialization/deserialization. As long as the source metadata remains unchanged between the ser/deser calls then this method will consistently produce the same ID.

I could also update `SetID` for this case to take (id string) as an argument which would override the field rather than calculate the hash.